Fix inconsistency in root key migration paragraph

lukpueh opened the pull request:
Fixes #56 

Updates an outdated paragraph in section 6.1 that explained how root keys are rotated by backwards-verifying signatures from the latest root file available on the repository back to the latest root file available
on the client.

The correct (new) workflow, which is also described in step 1 of the client application workflow of section 5, performs verification in a forward-directed manner, i.e. starting at the latest root on the client and going forward to the latest root on the repo.

mnm678 commented on tuf-spec.md:
```diff
@@ -1292,22 +1292,23 @@ non-volatile storage as FILENAME.EXT.
+   For example, if there is a 1.root.json that has threshold 2 and a
+   2.root.json that has threshold 3, 2.root.json MUST be signed by at least 2
+   keys defined in 1.root.json and at least 3 keys defined in 2.root.json.
+   The client starts the root keys update update process with the latest
```
```suggestion
   The client starts the root key update process with the latest
```

trishankatdatadog commented on tuf-spec.md:
```diff
@@ -1292,22 +1292,23 @@ non-volatile storage as FILENAME.EXT.
+
+   Clients that have outdated root keys can update to the latest set of trusted
+   root keys, by incrementally downloading all intermediate root metadata
+   files, and verifying that each root metadata at a given version is signed by
```
```suggestion
   files, and verifying that each current version of the root metadata is signed by
```

trishankatdatadog commented on tuf-spec.md:
```diff
@@ -1292,22 +1292,23 @@ non-volatile storage as FILENAME.EXT.
+   Clients that have outdated root keys can update to the latest set of trusted
+   root keys, by incrementally downloading all intermediate root metadata
+   files, and verifying that each root metadata at a given version is signed by
+   a threshold of keys defined in the previous version of root, and a threshold
```
```suggestion
   a threshold of keys specified by its immediate predecessor as well as a threshold
```

trishankatdatadog commented on tuf-spec.md:
```diff
@@ -1292,22 +1292,23 @@ non-volatile storage as FILENAME.EXT.
+   root keys, by incrementally downloading all intermediate root metadata
+   files, and verifying that each root metadata at a given version is signed by
+   a threshold of keys defined in the previous version of root, and a threshold
+   of keys defined in that given version of root.
```
```suggestion
   of keys specified by itself.
```

lukpueh commented:
Thanks for your typo fix and rewordings, @mnm678 and @trishankatdatadog. I squashed them into one commit and added another one that mentions the possibility and limits of a freeze attack. Let me know what you think.

trishankatdatadog commented on tuf-spec.md:
```diff
@@ -1292,22 +1292,28 @@ non-volatile storage as FILENAME.EXT.
+   2.root.json that has threshold 3, 2.root.json MUST be signed by at least 2
+   keys defined in 1.root.json and at least 3 keys defined in 2.root.json. The
+   client starts the root key update process with the latest version of root
+   metadata available on the client, and stops when no newer root metadata is
```
```suggestion
   metadata available on the client, and stops when no version N+1 (where N is the latest trusted version) of the root metadata is
```

trishankatdatadog commented:
@lukpueh Sorry, I have one more suggestion. Other than that, it's great, thank you so much!

lukpueh commented:
It's a fine suggestion, @trishankatdatadog, thanks! Let me quickly re-fold the lines, squash the UI commit and force push.
